picamera2/picamera2.py

```diff
@@ -44,6 +46,66 @@ class Preview(Enum):
     QTGL = 3
 
 
+class CameraManager:
+    def __init__(self):
+        self.cameras = {}
+        self.setup()
+        self._lock = threading.Lock()
+
+    def setup(self):
+        self.cms = libcamera.CameraManager.singleton()
+        self.thread = threading.Thread(target=self.listen, daemon=True)
+        self.running = True
+        self.thread.start()
+
+    def add(self, index, camera):
+        with self._lock:
+            self.cameras[index] = camera
+            if not self.running:
+                self.setup()
+
+    def cleanup(self, index):
+        flag = False
+        with self._lock:
+            del self.cameras[index]
+            if self.cameras == {}:
+                self.running = False
+                flag = True
+        if flag:
+            self.thread.join()
+
+    def listen(self):
+        import selectors
+        sel = selectors.DefaultSelector()
+        sel.register(self.cms.event_fd, selectors.EVENT_READ, self.handle_request)
+
+        while self.running:
+            events = sel.select(0.2)
+            for key, _ in events:
+                callback = key.data
+                callback()
+
+        sel.unregister(self.cms.event_fd)
+        self.cms = None
+
+    def handle_request(self, flushid=None):
+        """Handle requests
```

Comment:
Just wondering whether the `flushid` is worth the trouble, given that we have to clear the `picam2._requests` in the `stop_()` call anyway?

Reply:
Yeah, I'm not sure, I added it just to save creating new CompletedRequests for that camera ID